psycopg/psycopg/connection.py

```diff
@@ -265,29 +265,36 @@ def rollback(self) -> None:
     def cancel_safe(self, *, timeout: float = 30.0) -> None:
         """Cancel the current operation on the connection.
 
+        :param timeout: raise a `~errors.CancellationTimeout` if sending
+            the timeout doesn't succeed within `timeout` seconds.
+
+        Note that a successful cancel attempt on the client is not a guarantee
+        that the server will successfully manage to cancel the operation.
+
         This is a non-blocking version of `~Connection.cancel()` which
         leverages a more secure and improved cancellation feature of the libpq,
         which is only available from version 17.
 
         If the underlying libpq is older than version 17, the method will fall
         back to using the same implementation of `!cancel()`.
 
-        :raises ~psycopg.errors.CancellationTimeout: If the cancellation did
-            not terminate within specified timeout.
         """
         if not self._should_cancel():
             return
 
         if capabilities.has_cancel_safe():
-            try:
-                waiting.wait_conn(
-                    self._cancel_gen(timeout=timeout), interval=_WAIT_INTERVAL
-                )
-            except Exception as ex:
-                logger.warning("couldn't try to cancel query: %s", ex)
+            waiting.wait_conn(
+                self._cancel_gen(timeout=timeout), interval=_WAIT_INTERVAL
+            )
         else:
             self.cancel()
 
+    def _try_cancel(self, *, timeout: float = 30.0) -> None:
```

Comment:
I would suggest to remove the default value for `timeout` as callers always pass a (different) value. (The other option would be to set the default value to the one that's actually used, i.e. `5.0`, but I find it clearer to have it set by the caller.)

Reply:
What I thought is that, as a default timeout, something in the 30-60s region seems ok, because it's pretty much just to avoid to hang forever. That's what end users would call by default.

As the timeout to "get out of troubles" in our connection, I thought I wouldn't want to add a pretty long operation by default (although it's long only if the situation is really bad...)

If you think it's unclear why we propose 30s to the user, but use 5s ourselves, it's probably ok to drop the 5s introduced in the callers and to use the default which I'd leave to 30a.

Reply:
Yes, 30s as a default for user-facing code is totally fine. But, IIUC, this timeout here concerns `_try_cancel()` which is not called (or no longer called) by `cancel_safe()`.

Reply:
Oh yeah, you are totally right. We can make 5s a default of _try_cancel().

Reply:
Fixed in f9c3d202495b34a7516480bfb9cebb93c1775301.